docs/src/modules/components/HookApiPage.js

```diff
@@ -0,0 +1,286 @@
+/* eslint-disable react/no-danger */
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { exactProp } from '@mui/utils';
+import { alpha, styled } from '@mui/material/styles';
+import Alert from '@mui/material/Alert';
+import Typography from '@mui/material/Typography';
+import { useTranslate, useUserLanguage } from 'docs/src/modules/utils/i18n';
+import HighlightedCode from 'docs/src/modules/components/HighlightedCode';
+import MarkdownElement from 'docs/src/modules/components/MarkdownElement';
+import AppLayoutDocs from 'docs/src/modules/components/AppLayoutDocs';
+import Ad from 'docs/src/modules/components/Ad';
+
+const Asterisk = styled('abbr')(({ theme }) => ({ color: theme.palette.error.main }));
+
+const Wrapper = styled('div')({
+  overflow: 'hidden',
+});
+const Table = styled('table')(({ theme }) => {
+  const contentColor =
+    theme.palette.mode === 'dark'
+      ? alpha(theme.palette.primaryDark[900], 1)
+      : 'rgba(255, 255, 255, 1)';
+  const contentColorTransparent =
+    theme.palette.mode === 'dark'
+      ? alpha(theme.palette.primaryDark[900], 0)
+      : 'rgba(255, 255, 255, 0)';
```

Comment:
This file is duplicated between MUI Core and MUI X. We are adding it a 3rd time. It could be good to normalize this. Maybe it's outside of the scope of this PR, not sure how, but I think we only need it once.

Reply:
> This file is duplicated between MUI Core and MUI X. We are adding it a 3rd time. It could be good to normalize this. Maybe it's outside of the scope of this PR, not sure how, but I think we only need it once.

Will fix it in this PR.

Reply:
Done, I've adjusted the API to work with both components & hooks API.

Reply:
cc @flaviendelangle you could probably use it in the X too.

Reply:
Ah ok, you abstracted the `PropretiesTable`, it helps 👍. I was thinking of going all the way to create a generic API component and have a JavaScript API to control its rendering.